packages/expo-notifications/plugin/src/withNotificationsIOS.ts

```diff
@@ -1,11 +1,67 @@
-import { ConfigPlugin, withEntitlementsPlist } from '@expo/config-plugins';
+import {
+  ConfigPlugin,
+  withEntitlementsPlist,
+  IOSConfig,
+  withXcodeProject,
+} from '@expo/config-plugins';
+import fs from 'fs-extra';
+import path from 'path';
 
-export const withNotificationsIOS: ConfigPlugin<{ mode: 'production' | 'development' }> = (
+import { NotificationsPluginProps } from './withNotifications';
+
+type XcodeProject = any;
```

Comment:
This is `import { XcodeProject } from '@expo/config-plugins';`, maybe you need to bump the v of config-plugins to get it?